`Development`: Reorganize setup documentation (#7323)
ls1intum/Artemis: docs/admin/setup/kubernetes.rst

Reviewer: IMO, the Kubernetes guide should stay in the dev setup guide because "this does not result in a production-ready deployment"

Author: Neither does it result in a working dev-deployment at the moment. I’d leave it in the admin section as a special deployment similar to the multi-instance Artemis setup. Maybe someone that is familiar with Kubernetes can extract relevant pointers about how to deploy Artemis in their cluster using this documentation whereas I don’t think any developer is going to set up their dev-env using Kubernetes.

I think the goal is to whenever https://github.com/ls1intum/artemis-helm is ready, completely replace this part of the setup documentation with an updated one anyway.

Kubernetes Setup
----------------

This section describes how to set up an environment deployed in Kubernetes.

.. note::

>    These steps were part of an experimental setup.
>    This does not result in a production-ready deployment.

Reviewer 2: ```suggestion
   These steps were part of an experimental setup and do not result in a production-ready deployment.
```
